sys/kern/vm_object.c

```diff
@@ -36,37 +36,40 @@ void vm_object_add_page(vm_object_t *obj, off_t offset, vm_page_t *pg) {
 
   pg->object = obj;
   pg->offset = offset;
+  refcnt_acquire(&pg->ref_counter);
 
-  WITH_MTX_LOCK (&obj->mtx) {
-    vm_page_t *it;
-    TAILQ_FOREACH (it, &obj->list, obj.list) {
-      if (it->offset > pg->offset) {
-        TAILQ_INSERT_BEFORE(it, pg, obj.list);
-        obj->npages++;
-        return;
-      }
-      /* there must be no page at the offset! */
-      assert(it->offset != pg->offset);
-    }
+  SCOPED_MTX_LOCK(&obj->mtx);
 
-    /* offset of page is greater than the offset of any other page */
-    TAILQ_INSERT_TAIL(&obj->list, pg, obj.list);
-    obj->npages++;
+  vm_page_t *it;
+  TAILQ_FOREACH (it, &obj->list, obj.list) {
+    if (it->offset > pg->offset) {
+      TAILQ_INSERT_BEFORE(it, pg, obj.list);
+      obj->npages++;
+      return;
+    }
+    /* there must be no page at the offset! */
+    assert(it->offset != pg->offset);
   }
+
+  /* offset of page is greater than the offset of any other page */
+  TAILQ_INSERT_TAIL(&obj->list, pg, obj.list);
+  obj->npages++;
```

Comment:
Something changed here or WITH_MTX_LOCK was replaced by SCOPED_MTX_LOCK? Please don't do that!!! It waste just time when someone wants to read it.

Reply:
Fixed.